compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -1397,19 +1397,30 @@ class JSCodeGen()(implicit ctx: Context) {
     } else /*if (isScalaJSDefinedJSClass(currentClassSym)) {
       genJSSuperCall(tree, isStat)
     } else*/ {
-      val superCall = genApplyMethodStatically(
-          genThis()(sup.span), sym, genActualArgs(sym, args))
-
-      // Initialize the module instance just after the super constructor call.
-      if (isStaticModule(currentClassSym) && !isModuleInitialized &&
-          currentMethodSym.get.isClassConstructor) {
-        isModuleInitialized = true
-        val className = encodeClassName(currentClassSym)
-        val thisType = jstpe.ClassType(className)
-        val initModule = js.StoreModule(className, js.This()(thisType))
-        js.Block(superCall, initModule)
+      if (sym.owner.isAllOf(Scala2x | Trait) && sym.isConstructor) {
+        // calling genApplyMethodStatically will genereate a method with one argument for `this`, so we do it manually a zero-argument method
```

Comment:
This comment is not clear to me, and does not seem to match the explanation in the pull request description. Could you elaborate? A good way to make things clear in these weird corner cases is to give an example of *what code reaches the back-end*, and what is the IR that we generate for that corner case. See for example the Scaladoc of `def genClosure`, where this is done well.